modules/@apostrophecms/i18n/index.js

```diff
@@ -818,6 +818,393 @@ module.exports = {
+              req,
+              'localize-batch-can-localize-error',
+              'The document type can\'t be localized or insufficient permissions',
+              {
+                id: doc._id,
+                type: doc.type,
+                title: doc.title,
+                relationship: false
+              }
+            );
+            if (reporting) {
+              reporting.failure();
+            }
+            return log;
+          }
+          if (!relatedOnly) {
+            docs.push(doc);
+          }
+
+          if (relatedTypes.size > 0) {
+            try {
+              await findRelatedDocs(req.clone({ mode: 'draft' }), {
+                doc,
+                schema: self.apos.modules[doc.type].schema,
+                relatedTypes,
+                memo: docs,
+                seen
+              });
+            } catch (e) {
+              self.logError(
+                req,
+                'localize-batch-related-error',
+                'Error finding related documents',
+                {
+                  id: doc._id,
+                  type: doc.type,
+                  title: doc.title,
+                  error: e.message,
+                  stack: e.stack.split('\n').slice(1).map(line => line.trim())
+                }
+              );
+              if (reporting) {
+                reporting.failure();
+              }
+              return log;
+            }
+          }
+
+          let hasError = false;
+          for (const item of docs) {
+            const manager = self.apos.doc.getManager(item.type);
+            // Not using Promise.allSettled because of potential
+            // rate limit issues with external services when i.e.
+            // automatically translating content.
+            // Related info:
+            // https://cookbook.openai.com/examples/how_to_handle_rate_limits
+            for (const locale of toLocales) {
+              const payload = {
+                _id: item._id,
+                aposDocId: item.aposDocId,
+                locale,
+                type: item.type,
+                title: item.title,
+                relationship: item.aposDocId !== doc.aposDocId ? doc.aposDocId : false,
+                error: false
+              };
+              try {
+                await manager.localize(req, item, locale, {
+                  update: !payload.relationship ? true : update,
+                  batch: true
+                });
+                log.push(payload);
+              } catch (e) {
+                hasError = true;
+                payload.error = e.name ?? true;
+                // This is the only detail that we know of.
+                // XXX A better way to handle data sent to the UI as a
+                // human-readable message is a standard error payload property.
+                // For example `error.data.detail`.
+                if (e.data?.parentNotLocalized) {
+                  payload.detail = 'apostrophe:parentNotLocalized';
```

Comment:
We may want to add `detail: 'apostrophe:parentNotLocalized'` where `parentNotLocalized` is calculated and assigned. This way we can remove the redundant checks here and in the i18n UI.